File: 프로그래머스/1/388351. 유연근무제/유연근무제.java
import java.util.*;
class Solution {
    public int solution(int[] schedules, int[][] timelogs, int startday) {
        int answer = 0;
        for(int i=0; i<schedules.length; i++){
            if(check(schedules[i], timelogs[i], startday)){
                answer++;
            }
        }
        return answer;
    }
    
    public static int changeTime(int time){
        int hour = time / 100;
        int minute = time % 100;
        int sum = hour*60 + minute;
        return sum;
    }
    
    public static boolean check(int time, int[] timelog, int startday){
        int needTime = changeTime(time) + 10;               // 출근해야 하는 시각
        int count = 0;
        for(int i=0; i<7; i++){
            int commuteTime = changeTime(timelog[i]);
            if(startday == 6 || startday == 7){
                startday++;
                if(startday > 7){
                    startday -= 7;
                }
                continue;
            }
            if(commuteTime > needTime){
                return false;
            }
            startday++;
            if(startday > 7){
                startday -= 7;
            }
        }
        return true;
    }
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    answer = 0
    
    def changeTime(time) :
        hour = time // 100;
        minute = time % 100;
        return hour * 60 + minute;
    
    def check(time, timelog, startday) :
        needTime = changeTime(time) + 10;
        for i in range(7) :
            commuteTime = changeTime(timelog[i]);
            if startday == 6 or startday == 7 :
                startday += 1;
                if startday > 7 :
                    startday -= 7;
                continue;
            if(commuteTime > needTime) :
                return False;
            startday += 1;
            if startday > 7 :
                startday -= 7;
        return True;
    
    for i in range(len(schedules)) :
        if(check(schedules[i], timelogs[i], startday)):
            answer += 1;
    return answer